src/function/scalar/date/date_diff.cpp

```diff
@@ -0,0 +1,333 @@
+#include "duckdb/function/scalar/date_functions.hpp"
+#include "duckdb/common/enums/date_part_specifier.hpp"
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/types/date.hpp"
+#include "duckdb/common/types/interval.hpp"
+#include "duckdb/common/types/time.hpp"
+#include "duckdb/common/types/timestamp.hpp"
+#include "duckdb/common/vector_operations/ternary_executor.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/storage/statistics/numeric_statistics.hpp"
+
+namespace duckdb {
+
+// This function is an implementation of the "period-crossing" date difference function from T-SQL
+// https://docs.microsoft.com/en-us/sql/t-sql/functions/datediff-transact-sql?view=sql-server-ver15
+struct DateDiff {
+	static constexpr const int32_t SECS_PER_WEEK = Interval::SECS_PER_DAY * Interval::DAYS_PER_WEEK;
```

Comment:
These constants are mostly gathered in the `Interval` class right now, perhaps nice to group them up either here or there? Or in another place?

Reply:
Looks like some of them got added already!